packages/dm-core-plugins/src/generic-list/GenericTablePlugin.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { ChangeEvent, useContext, useEffect, useState } from 'react'
+import {
+  AuthContext,
+  DmssAPI,
+  ErrorResponse,
+  IUIPlugin,
+  Loading,
+  TGenericObject,
+  useDocument,
+} from '@development-framework/dm-core'
+import { AxiosError, AxiosResponse } from 'axios'
+import { Button, Icon, Input, Progress, Table } from '@equinor/eds-core-react'
+import { delete_to_trash, library_add } from '@equinor/eds-icons'
+
+type TGenericTablePlugin = {
+  editMode: boolean
+  columns: string[]
+  showDelete: boolean
+}
+const defaultConfig: TGenericTablePlugin = {
+  editMode: true,
+  columns: ['name', 'type'],
+  showDelete: true,
+}
+
+export const GenericTablePlugin = (
+  props: IUIPlugin & { config?: TGenericTablePlugin }
+) => {
```

Comment:
What if we made a new generic type for this? ie
```suggestion
type TUIPlugin<TConfig> = {
  type: string
  idReference: string
  onSubmit?: (data: any) => void
  onOpen?: (data: any) => void
  config?: TConfig
}
export const GenericTablePlugin = (props: TUIPlugin<TGenericTablePlugin>)
```

Reply:
Absolutely. Agreed.
However, cant make that change in this PR, as that would cause tests to fail as "dm-core-packages" needs to pass with latest published version of "dm-core".